libcontainer/process_linux.go

```diff
@@ -836,6 +840,28 @@ func (p *initProcess) createNetworkInterfaces() error {
 	return nil
 }
 
+// setupNetworkDevices sets up and initializes any defined network interface inside the container.
+func (p *initProcess) setupNetworkDevices() error {
+	// host network pods does not move network devices.
+	if !p.config.Config.Namespaces.Contains(configs.NEWNET) {
+		return nil
+	}
+	// get the namespace defined by the config and fall back
+	// to the one created by runc to run the container process.
+	nsPath := p.config.Config.Namespaces.PathOf(configs.NEWNET)
+	if nsPath == "" {
+		nsPath = fmt.Sprintf("/proc/%d/ns/net", p.pid())
+	}
+	for name, netDevice := range p.config.Config.NetDevices {
+		err := devChangeNetNamespace(name, nsPath, *netDevice)
+		if err != nil {
+			return err
```

Comment:
I'd wrap the error, so we don't lose context and know where it comes from. Something like:
```suggestion
			return fmt.Errorf("move netDevice: %w", err)
```